code-gurus/mhsat-api
# Add delete functionality

# PR Details
 - allow the user to delete full test if exists
- delete single questions if both the test and the question exist on the database.



## Description
 - A user can delete a Test by Id only if it exists
 - A user can delete a Question only if its Test id in exists and the question id exists


## Related Issue






## Motivation and Context


 - Allow deleting of resources
## How Has This Been Tested





1. Pull and checkout to  branch `ft-delete-test`
2. Run server with `npm run dev`
3. Try and delete both and existing and none existing test using the route below:
    - `http://localhost:8000/api/v1/tests/:id`
4. Try and delete both and existing and none existing question using the route below:
    - `http://localhost:8000/api/v1/tests/:id/questions/:questionId`
5. Expect relevant error messages.
## Types of changes




## Checklist





## Screenshots


<img width="1440" alt="Screen Shot 2020-02-20 at 12 21 37" src="https://user-images.githubusercontent.com/22627452/74924725-9a4e7e00-53db-11ea-8480-b2f9bd32e6dc.png">
<img width="1440" alt="Screen Shot 2020-02-20 at 12 22 10" src="https://user-images.githubusercontent.com/22627452/74924751-a6d2d680-53db-11ea-8dde-d0d22cc4ede1.png">
